### DISCUSSION TITLE
Clarification question re Installation wiki

### DISCUSSION THREAD

**Discussion by seamusdemora** [Arkq/bluez-alsa#708] [Q&A] [answered]
Sorry, but I'm confused on a couple of points:

What I've done so far:

```shell
cd; cd bluez-alsa
git pull origin master
autoreconf --install --force
...
``` 

My confusion begins with Step 2 of the [installation wiki](https://github.com/arkq/bluez-alsa/wiki/Installation-from-source) where it says: **"Create a new directory in which to perform the build."**

1. If I've built it before (i.e. this is an update/upgrade), should I use the existing `~/bluez-alsa/build` - or should I create a new build folder (i.e. `~/bluez-alsa/build2`) - _or something else_??
    
2. Do I need to re-run `configure` with the options I used previously - or will they remain in force for this re-build?

**Comment by arkq** [comment, marked as answer]
> 1. If I've built it before (i.e. this is an update/upgrade), should I use the existing ~/bluez-alsa/build - or should I create a new build folder (i.e. ~/bluez-alsa/build2) - or something else??

You can use the existing one, or you can create a new one, it doesn't really matter (in 99% cases :D). However, in case of bluez-alsa upgrade from previous release to the new one (in this particular case), when using the old build directory, you will have to run `make clean` in it, before running `make`. There are some issues with autotools (or with configuration in bluez-alsa project...).

> 2. Do I need to re-run configure with the options I used previously - or will they remain in force for this re-build?

When using previous build directory, the (re)configuration should be auto-triggered by the build system, so you do not have to run it manually (but you can).

The simplest upgrade step might look like:

```sh
~/bluez-alsa/build $ make clean
~/bluez-alsa/build $ cd ..
~/bluez-alsa $ git pull
~/bluez-alsa $ autoreconf --install --force
~/bluez-alsa $ cd build
~/bluez-alsa/build $ ../configure <previously used options> --enable-new-feature # (optional step, to enable/disable new features added in a new release)
~/bluez-alsa/build $ make
~/bluez-alsa/build $ sudo make install
```

**Reply by seamusdemora** [reply]
This is great - many thanks! 

And yet _another_ question: I may have mistakenly entered one of the `configure` options based on [an earlier question](https://github.com/arkq/bluez-alsa/discussions/695#discussioncomment-9154371): 

Which is the correct command: 
```
../configure --enable-systemd --bluealsa-cli 

# OR...

../configure --enable-systemd --enable-cli
``` 

After re-reading the wiki, it seems like it is `--enable-cli`, but from my history file, when I built this the first time, I apparently entered `--bluealsa-cli` instead! 

_**ALSO:**_ Any advantage to be had for me to use the `configure` option `--with-bluealsauser=USER`?  My platform is Raspberry Pi Zero 2W & `pi` is the only user on the system (also a member of group `bluetooth`)

**Reply by arkq** [reply]
> After re-reading the wiki, it seems like it is --enable-cli, but from my history file, when I built this the first time, I apparently entered --bluealsa-cli instead!

Using wrong option will do nothing (of course it will not enable desired feature). If you want to compile different configuration, then do `make clean` and then reconfigure project with `../configure <new options>`. Then make and install. That's it.

> Any advantage to be had for me to use the configure option --with-bluealsauser=USER? My platform is Raspberry Pi Zero 2W & pi is the only user on the system (also a member of group bluetooth)

In your case there are not benefits from this. This option is for running bluealsa system service as a given user instead of `root`. Just keep it simple. Default should be good enough.
